Tests/QuranAudioKitTests/DownloadedRecitersServiceTests.swift

```diff
@@ -0,0 +1,64 @@
+//
+//  DownloadedRecitersServiceTests.swift
+//  
+//
+//  Created by Zubair Khan on 1/2/23.
+//
+
+import Foundation
+@testable import QuranAudioKit
+import XCTest
+
+class DownloadedRecitersServiceTests: XCTestCase {
+    private var service: DownloadedRecitersService!
+    private var fileManager = FileManager.default
+
+    override func setUpWithError() throws {
+        service = DownloadedRecitersService()
+    }
+
+    func testFilterDownloadedReciters() {
+        let reciter1 = createReciter(id: 1)
+        let reciter2 = createReciter(id: 2)
+        let reciter3 = createReciter(id: 3)
+        let reciter4 = createReciter(id: 4)
+        let reciter5 = createReciter(id: 5)
+        let allReciters = Array(arrayLiteral: reciter1, reciter2, reciter3, reciter4, reciter5)
+
+        createReciterDir(reciter2)
+        createReciterDir(reciter4)
+        createReciterDir(reciter5)
+
+        // Reciter2's directory will be empty, therefore it shouldn't be considered a downloaded reciter
+        createReciterFile(reciter4)
+        createReciterFile(reciter5)
+
+        let downloadedReciters = service.downloadedReciters(allReciters)
+        XCTAssertEqual(downloadedReciters, [reciter4, reciter5])
+    }
+
+    private func createReciterDir(_ reciter: Reciter) {
+        try? fileManager.createDirectory(at: URL(string: reciterDirPath(reciter))!, withIntermediateDirectories: true)
+    }
+
+    private func createReciterFile(_ reciter: Reciter) {
+        let fileURL = URL(fileURLWithPath: reciterDirPath(reciter)).appendingPathComponent("surah1")
+        try? "testing123".write(to: fileURL, atomically: true, encoding: .utf8)
+    }
+
+
+    private func reciterDirPath(_ reciter: Reciter) -> String {
```

Comment:
You don't need this function, you should be able to use Reciter.path or Reciter.localFolder()